js/components/grid/course_panel.js.jsx

```diff
@@ -122,13 +125,23 @@ class Course extends React.Component {
     this.props.removeCourse(this.props.courseCode);
   }
 
+  setModalRef(element) {
+    this.modal = element;
+  }
+
+  selectInfo() {
+    this.modal.openModal(this.state.courseInfo.courseCode.substring(0, 6));
+  }
+
   render() {
     return (
       <li key={this.props.courseCode} id={this.props.courseCode + "-li"} className={"ui-accordion ui-widget ui-helper-reset"}>
         <div className="ui-accordion-header ui-helper-reset ui-state-default ui-accordion-icons ui-accordion-header-active ui-state-active ui-corner-top"
               id={"ui-accordion-" + this.props.courseCode + "-li-header-0"}>
+          <Modal ref={this.setModalRef}/>
```

Comment:
It's more common to see `ref`s set using anonymous functions (i.e., `r => this.modal = r`)